api/src/config.rs

```diff
@@ -133,13 +139,24 @@ impl BootloaderConfig {
             },
         );
 
-        concat_106_9(
+        let buf = concat_106_9(
             buf,
             match minimum_framebuffer_width {
                 Option::None => [0; 9],
                 Option::Some(addr) => concat_1_8([1], addr.to_le_bytes()),
             },
-        )
+        );
+
+        let log_level: u8 = match log_level {
+            log::LevelFilter::Trace => 0,
+            log::LevelFilter::Debug => 1,
+            log::LevelFilter::Info => 2,
+            log::LevelFilter::Warn => 3,
+            log::LevelFilter::Error => 4,
+            log::LevelFilter::Off => 5,
+        };
```

Comment:
The `LevelFilter` enum is `repr(usize)`, so I don't think that it's a good idea to use a different integer representation. Instead, we should use the canonical one:

```suggestion
        let log_level = u8::try_from(log_level as usize).unwrap();
```

If we define our own `LevelFilter` enum, we could even make it `repr(u8)` to avoid the `u8::try_from`.